### ISSUE TITLE
Association Cache incompatibility with ONLY_FULL_GROUP_BY

### ISSUE THREAD

**Issue by damianhxy** [autolab/Autolab#1961]
Either resolve the incompatibility, or remove the association cache altogether (is it still necessary?)

```
--- !ruby/exception:ActiveRecord::StatementInvalid
message: 'Mysql2::Error: Expression #1 of ORDER BY clause is not in SELECT list, references
column ''autolab.assessments.due_at'' which is not in SELECT list; this is incompatible
with DISTINCT'
backtrace:
...
- "/usr/local/lib/ruby/gems/2.7.0/gems/mysql2-0.4.10/lib/mysql2/client.rb:119:in `query'"
- "/usr/local/lib/ruby/gems/2.7.0/gems/activerecord-6.0.5/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:201:in
`block (2 levels) in execute'"
...
- "/usr/local/lib/ruby/gems/2.7.0/gems/lockbox-1.0.0/lib/lockbox/calculations.rb:4:in
`pluck'"
- "/opt/autolab/app/models/course.rb:257:in `assessment_categories'"
- "/opt/autolab/app/models/grade_matrix.rb:103:in `block in matrix!'"
- "/usr/local/lib/ruby/gems/2.7.0/gems/activerecord-6.0.5/lib/active_record/relation/delegation.rb:88:in
`each'"
- "/usr/local/lib/ruby/gems/2.7.0/gems/activerecord-6.0.5/lib/active_record/relation/delegation.rb:88:in
`each'"
- "/opt/autolab/app/models/grade_matrix.rb:93:in `matrix!'"
- "/opt/autolab/app/models/grade_matrix.rb:19:in `block in initialize'"
...
- "/usr/local/lib/ruby/gems/2.7.0/gems/activesupport-6.0.5/lib/active_support/cache.rb:332:in
`fetch'"
- "/opt/autolab/app/models/grade_matrix.rb:11:in `initialize'"
- "/opt/autolab/app/controllers/gradebooks_controller.rb:20:in `new'"
- "/opt/autolab/app/controllers/gradebooks_controller.rb:20:in `view'"
- "/usr/local/lib/ruby/gems/2.7.0/gems/actionpack-6.0.5/lib/action_controller/metal/basic_implicit_render.rb:6:in
`send_action'"
...
```

Stack Dump
```
/opt/autolab/app/models/course.rb:257:in `assessment_categories'
/opt/autolab/app/models/grade_matrix.rb:103:in `block in matrix!'
/opt/autolab/app/models/grade_matrix.rb:93:in `matrix!'
/opt/autolab/app/models/grade_matrix.rb:19:in `block in initialize'
/opt/autolab/app/models/grade_matrix.rb:11:in `initialize'
/opt/autolab/app/controllers/gradebooks_controller.rb:20:in `new'
/opt/autolab/app/controllers/gradebooks_controller.rb:20:in `view'
```

**Comment by damianhxy** [comment]
More context by Chaskiel:

The 'assessments' returned by the cache are: (from association_cache.rb#L133)
    @assessments = cache.sorted_assessments if cache.sorted_assessments
When cache.load_assessments is used (by the gradebook..., on association_cache.rb#L26)
    @sorted_assessments = @course.assessments.ordered.load

**Comment by damianhxy** [comment]
The incompatibility was resolved in #1962, it remains to see if the cache can be removed

**Comment by damianhxy** [comment]
Won't fix - Dave and Greg have shared that Association Cache was instrumental to speeding up performance.

Tested that the current Association Cache works well.
